src/nl/rubensten/texifyidea/ui/tablecreationdialog/TableCreationDialogWrapper.kt

```diff
@@ -0,0 +1,191 @@
+package nl.rubensten.texifyidea.ui.tablecreationdialog
+
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.ui.ToolbarDecorator
+import com.intellij.ui.components.JBLabel
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.table.JBTable
+import nl.rubensten.texifyidea.action.tablewizard.TableInformation
+import java.awt.BorderLayout
+import java.awt.Dimension
+import java.awt.event.ActionEvent
+import javax.swing.*
+import javax.swing.border.EmptyBorder
+
+/**
+ * Wrapper that contains the table creation dialog. It validates the form when clicking the OK button.
+ *
+ * @param columnTypes The types of the columns of the table, see [ColumnType], always start with an empty table.
+ * @param tableModel The model of the table, always start with an empty table.
+ * @param tableInformation Information about the table that is needed to convert it to latex.
+ *
+ * UI components that have to be validated when clicking the OK button, i.e., checking if the user entered something.
+ * @param table The JTable component that shows the table.
+ * @param caption The text field that contains the caption for the table.
+ * @param reference The text field that contains the label for the table. It has a default value "tab:" to encourage usage
+ * of label conventions.
+ *
+ * @author Abby Berkers
+ */
+class TableCreationDialogWrapper(private val columnTypes: MutableList<ColumnType> = emptyList<ColumnType>().toMutableList(),
+                                 private val tableModel: TableCreationTableModel = TableCreationTableModel(),
+                                 var tableInformation: TableInformation = TableInformation(tableModel, columnTypes, "", ""),
+                                 // Components that have to be validated when clicking the OK button.
+                                 private val table: JTable = JBTable(tableModel),
+                                 private val caption: JBTextField = JBTextField(),
+                                 private val reference: JBTextField = JBTextField("tab:"))
+    : DialogWrapper(true) {
+
+    init {
+        // Initialise the dialog, otherwise it shows as a line (i.e., infinitely small) and without any of the elements.
+        init()
+    }
+
+    /**
+     * Add a table column.
+     *
+     * @param title of the column.
+     * @param typedColumnIndex is the column type of the column.
+     */
+    private val addColumnFun = fun(title: String, typedColumnIndex: Int, _: Int) {
```

Comment:
Could be a method? Or am I missing something obvious here why it should be a property?

Reply:
This is an anonymous function that is passed to the `TableCreationEditColumnDialog`, it's the function that gets invoked when clicking OK.